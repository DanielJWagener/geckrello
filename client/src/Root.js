// Simply creates a Redux store and passes it in to a provider tag.
// This logic was originally in index.js, but we've movied it here to
// make our testing process easier.
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  {},
  composeEnhancers(applyMiddleware(reduxThunk))
);

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
