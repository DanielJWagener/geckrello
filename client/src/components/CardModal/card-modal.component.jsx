import React from "react";
import ReactDOM from "react-dom";
import CardDescription from "../CardDescription/card-description.component";
import Checklist from "../Checklist/checklist.component";
import CardUtilities from "../CardUtilities/card-utilities.component";
import history from "../../utilities/history";

import "./card-modal.styles.scss";

function CardModal(props) {
  return ReactDOM.createPortal(
    <div
      onClick={() => history.goBack()}
      className="modal__overlay"
      id={props.cardId}
    >
      <div onClick={e => e.stopPropagation()} className="modal">
        <div className="modal__heading">
          <h2 className="modal__heading--main">{props.cardTitle}</h2>
          <h4 className="modal__heading--sub">
            In list: <span className="list-title">{props.inList}</span>
          </h4>
        </div>

        <a href="#board" className="modal__close" draggable="false">
          &times;
        </a>
        <div className="modal__main">
          <div className="modal__main--left">
            <CardDescription cardId={props.cardId} />
            <Checklist cardId={props.cardId} />
          </div>
          <div className="modal__main--right">
            <CardUtilities cardId={props.cardId} />
          </div>
        </div>
      </div>
    </div>,
    document.querySelector("#modal")
  );
}

export default CardModal;
