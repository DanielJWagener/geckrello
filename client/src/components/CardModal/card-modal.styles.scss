@import "../../sass/abstracts/_variables.scss";

.modal {
  width: 50%;
  min-height: 40rem;
  max-height: calc(var(--vh, 1vh) * 100);
  overflow-y: auto;
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  background-color: $component-background;
  border-radius: 0.5rem;
  padding: 1rem;

  @media only screen and (max-width: 56.25em) {
    width: 75%;
  }

  @media only screen and (max-width: 37.5em) {
    width: 90%;
  }

  &__overlay {
    position: fixed;
    top: 0;
    left: 0;
    height: 100vh;
    width: 100%;
    background-color: rgba(0, 0, 0, 0.8);
    z-index: 9999;
    visibility: hidden;
    opacity: 0;
    transition: all 0.3s;

    &:target {
      visibility: visible;
      opacity: 1;
    }
  }

  &__heading {
    text-align: center;
    margin-bottom: 3rem;

    &--main {
      font-size: 3rem;
    }

    &--sub {
      font-size: 1.5rem;

      .list-title {
        text-decoration: underline;
      }
    }
  }

  &__close {
    position: absolute;
    top: 3px;
    right: 1rem;
    cursor: pointer;
    font-size: 3rem;
    text-decoration: none;
    color: $default-font-color;

    &:hover {
      color: red;
    }
  }

  &__widget-heading {
    font-size: 2rem;
    margin-bottom: 0.5rem;
    display: inline-block;
    margin-right: 1rem;
  }

  &__button--inline {
    display: inline-block;
    background-color: #888;
    color: white;
    font-size: 1.5rem;
    padding: 0.5rem 1rem 0.5rem 1rem;
    margin-bottom: 0.5rem;
    border-radius: 0.5rem;
    border: #888;
    cursor: pointer;

    &:hover {
      background-color: #444;
    }
  }

  &__body {
    width: 100%;
    display: grid;
    grid-template-columns: [modal-body-start main-start] 75% [main-end utilities-start] 25% [utilities-end modal-body-end];

    &--main {
      grid-column: modal-body-start / main-end;
      @media only screen and (max-width: 37.5em) {
        width: 100%;
      }
    }

    &--utilities {
      grid-column: utilities-start / modal-body-end;
      @media only screen and (max-width: 37.5em) {
        width: 100%;
      }
    }
  }
}
