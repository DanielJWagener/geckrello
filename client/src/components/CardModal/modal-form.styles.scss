@import "../../sass/abstracts/_variables.scss";

.modal-form {
  padding: 0.5rem;

  &__group:not(:last-child) {
    margin-bottom: 0.5rem;
  }

  &__input {
    display: block;
    width: 100%;
    font-size: 1.6rem;
    line-height: 2.4rem;
    padding: 0 1rem;
    border-radius: 0.5rem;
    border: none;
    color: #333;

    &--validation {
      border-bottom-style: solid;
      border-bottom-width: 2px;
      border-bottom-color: #555;
    }
  }

  &__textarea {
    font-family: inherit;
    display: block;
    background-color: white;
    width: 100%;
    height: 4rem;
    padding: 0.5rem;
    border-radius: 5px;
    border: none;
    border-bottom: 2px solid #555;
  }

  &__submit,
  &__edit {
    display: inline-block;
    padding: 0.5rem 1rem;
    border-radius: 0.5rem;
    border: none;
    font-size: 1.6rem;
    background-color: $color-secondary;
    color: white;
    cursor: pointer;
    transition: all 0.2s;

    &:focus,
    &:active {
      outline: none;
      transform: translateY(2px);
    }

    &:hover {
      background-color: $color-secondary-dark;
    }
  }

  &__cancel {
    display: inline-block;
    margin-left: 1rem;
    font-size: 1.5rem;
    text-decoration: none;
    color: #333;
    background-color: transparent;
    border: none;
    cursor: pointer;

    &:hover {
      color: red;
    }
  }
}
