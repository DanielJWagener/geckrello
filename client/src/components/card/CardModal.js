import React from "react";
import Description from "./Description";
import Checklist from "./Checklist";
import SideMenu from "./SideMenu";

function CardModal(props) {
  return (
    <div className="modal__overlay" id={props.cardId}>
      <div className="modal">
        <div className="modal__heading">
          <h2 className="modal__heading--main">{props.cardTitle}</h2>
          <h4 className="modal__heading--sub">
            In list: <span className="list-title">{props.inList}</span>
          </h4>
        </div>

        <a href="#board" className="modal__close" draggable="false">
          &times;
        </a>
        <div className="modal__main">
          <div className="modal__main--left">
            <Description cardId={props.cardId}></Description>

            <Checklist cardId={props.cardId}></Checklist>
          </div>
          <div className="modal__main--right">
            <SideMenu cardId={props.cardId}></SideMenu>
          </div>
        </div>
      </div>
    </div>
  );
}

export default CardModal;
