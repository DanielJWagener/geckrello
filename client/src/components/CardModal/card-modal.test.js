import { shallow } from "enzyme";
import React from "react";
import CardModal from "./card-modal.component";
import { CardModal as CardModalPure } from "./card-modal.component";
import Root from "../../Root";

it("renders CardModal component", () => {
  expect(
    shallow(<CardModal history={{ goBack: () => {} }} />).debug()
  ).toMatchSnapshot();
});
