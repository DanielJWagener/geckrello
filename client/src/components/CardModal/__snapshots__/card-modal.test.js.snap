// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CardModal component 1`] = `
"<ContextConsumer>
  [function]
</ContextConsumer>"
`;
